ldcbench.http-node/src/main/java/org/dice_research/ldcbench/nodes/http/simple/SimpleHttpServerComponent.java

```diff
@@ -174,12 +179,16 @@ protected Container createContainer() throws Exception {
         String[] accessUriTemplates = Stream.of(nodeMetadata).map(nm -> nm.getAccessUriTemplate())
                 .toArray(String[]::new);
         if (dumpFileNode) {
-            resource = DumpFileResource.create(cloudNodeId.get(),
-                    Stream.of(nodeMetadata).map(nm -> nm.getResourceUriTemplate()).toArray(String[]::new),
-                    Stream.of(nodeMetadata).map(nm -> nm.getAccessUriTemplate()).toArray(String[]::new),
-                    graphs.toArray(new Graph[graphs.size()]),
-                    r -> r.getPath().toString().equals(dumpFilePath),
-                    dumpFileLang, dumpFileCompression, dumpfileArchiver);
+        	for(int i=0; i<noOfGraphs;i++) {
+        	   resource = DumpFileResource.create(cloudNodeId.get(),
+                       Stream.of(nodeMetadata).map(nm -> nm.getResourceUriTemplate()).toArray(String[]::new),
+                       Stream.of(nodeMetadata).map(nm -> nm.getAccessUriTemplate()).toArray(String[]::new),
+                       graphs.toArray(new Graph[graphs.size()]),
+                       r -> r.getPath().toString().equals(dumpFilePath),
+                       dumpFileLang, dumpFileCompression,multipleFiles,dumpfileArchiver);
+        	   
+        	   resources.add(resource);
+        	}
```

Comment:
Just encapsulating this piece of code into a loop is not a good idea. Please have a look at these lines that generate several objects. For example, `graphs.toArray(new Graph[graphs.size()])` creates a new array. Since the lines simply have been encapsulated into a the for loop, the array would be generated again and again in each iteration.

Instead, these objects should be stored in local variables and reused when necessary.

Please also think about the variable `multipleFiles`. This flag is created based on the attribute `noOfGraphs` and is also stored as attribute. I do not see a reason why the flag has to be an attribute if it is used only in this single place and could be easily a local variable.